airflow/www/templates/airflow/dag.html

```diff
@@ -204,7 +204,12 @@ <h4 class="pull-right" style="user-select: none;-moz-user-select: auto;">
                   <input type="hidden" name="csrf_token" value="{{ csrf_token() }}">
                   <input type="hidden" name="dag_id" value="{{ dag.dag_id }}">
                   <input type="hidden" name="unpause" value="True">
-                  <input type="hidden" name="origin" value="{{ url_for(request.endpoint, dag_id=dag.dag_id) }}">
+                  <!-- for task instance detail pages, dag_id is still a query param -->
+                  {% if request.args.get('dag_id') is not none %}
```

Comment:
Not sure if this works…

```suggestion
                  {% if 'dag_id' in request.args %}
```